_posts/2025-09-15-quarkus-a2a-multi-agent-content-creation.adoc

```diff
@@ -0,0 +1,244 @@
+      "url": "http://localhost:10002"
+    }
+  ],
+  "preferredTransport": "JSONRPC",
+  "protocolVersion": "0.3.0"
+}
+----
+
+[NOTE]
+====
+The agent card for an A2A server agent can be found using its Well-Known URI, e.g., for the Content Writer agent, we can fetch its agent card using http://localhost:10002/.well-known/agent-card.json.
+====
+
+The description and skills specified in the Content Writer's agent card allow the Host agent's LLM to determine that the writing sub-task should be sent to the Content Writer agent.
+
+==== Agent Communication
+
+The Host agent communicates with each A2A server agent using an A2A client. Notice that an A2A client
+does not need to be written in the same programming language as an A2A server since all that matters is
+that both are using the A2A protocol to communicate with each other.
+
+Later on in this post, we'll see how we can easily swap out the TypeScript Content Editor agent for
+an equivalent agent written in Java, highlighting the flexibility and interoperability that's made
+possible by the A2A protocol.
+
+=== Running the Agents
+
+Let's get this multi-agent system running locally.
+
+==== Content Planner
+
+In a terminal:
+
+[source,shell]
+----
+cd samples/python/agents/content_planner
+----
+
+Follow the instructions in the `content_planner` https://github.com/a2aproject/a2a-samples/blob/main/samples/python/agents/content_planner/README.md[README.md] to start the Content Planner agent.
+
+==== Content Writer
+
+In a terminal:
+
+[source,shell]
+----
+cd samples/java/agents/content_writer
+----
+
+Follow the instructions in the `content_writer` https://github.com/a2aproject/a2a-samples/blob/main/samples/java/agents/content_writer/README.md[README.md] to start the Content Writer agent.
+
+==== Content Editor
+
+In a terminal:
+
+[source,shell]
+----
+cd samples/js/src/agents/content-editor
+----
+
+Follow the instructions in the `content-editor` https://github.com/a2aproject/a2a-samples/blob/main/samples/js/src/agents/content-editor/README.md[README.md] to start the Content Editor agent.
+
+==== Host
+
+In a terminal:
+
+[source,shell]
+----
+cd samples/python/hosts/content_creation
+uv run .
+----
+
+[NOTE]
+====
+As mentioned in the agent `README.md` files, don't forget to create a `.env` file for each agent with your
+`GOOGLE_API_KEY`. This is needed since the agents in this sample make use of Gemini. You can create a
+Google AI Studio API Key for free https://aistudio.google.com/[here].
+====
+
+=== Access the Content Creation Application
+
+Now that all of our agents are up and running, from your browser, navigate to http://localhost:8083.
+
+Try asking questions like:
+
+* Create a short, concise LinkedIn post about getting started with the Agent2Agent protocol
+* Create a short, upbeat X post about Quarkus LangChain4j
+
+image::UI.png[scaledwidth=100%]
+
+=== Swap Out an Agent
+
+One of the most powerful features of the A2A protocol is its interoperability. Let's see this in
```

Comment:
You forgot to mention that the agents must be shut down before going there.

Reply:
Made this more clear.